### PR TITLE
Improve inkscape compatibility 

### PR DESCRIPTION
**Pull request by kasbah**
Inkscape has some issues rendering the SVGs produced due to expecting groups within `mask` I guess.

Before:

After:

There is still a problem as the copper-finish  isn't rendered properly (should be golden but is invisible) which I haven't managed to fix yet.

### CODE REVIEW

**Comment by mcous** [comment]
Thanks, this looks cool. Although this does seem like an Inkscape bug, more compatibility it good! A couple things:
* Can you make a few style fixes? I'll comment on them directly
* In the future, please make a feature branch for PRs, as described [here](http://yangsu.github.io/pull-request-tutorial/)
* Maybe check out using a [clipPath](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/clipPath) for the finish
    * Right now, the finish is formed by masking the soldermask with the copper
    * It could work if instead, we clip the copper with the soldermask

File: src/build-board.coffee [hunk 1]
```diff
@@ -166,10 +166,10 @@ module.exports = (name, layers = []) ->
     # now build group for the color and the silkscreen (if it exists) and
     # mask away the soldermask holes
     smId = "#{name}-sm_#{uniqueId()}"
-    defs.push { mask: { id: smId, color: '#000', _: [
+    defs.push { mask: { id: smId, color: '#000', _: { g: { _: [
           bboxRect null, '#fff'
           { use: { 'xlink:href': "##{mask}" } }
-        ]
+        ]}}
```
Comment (mcous) [inline src/build-board.coffee]: Please try to match the indent style for braces and brackets with the rest of the file

File: src/build-board.coffee [hunk 2]
```diff
@@ -194,11 +194,11 @@ module.exports = (name, layers = []) ->
   mechId = null
   if drill.length || edgeBbox?
     mechId = "#{name}-mech_#{uniqueId()}"
-    mechMask = { mask: { id: mechId, color: '#000', _: [] } }
-    mechMask.mask._.push if edgeBbox? then { use: { 'xlink:href': "##{edge}" } } else bboxRect null, '#fff'
-    mechMask.mask._.push { use: { 'xlink:href': "##{d}" } } for d in drill
+    mechMask = { mask: { id: mechId, color: '#000', _: { g:{ _: [] } } } }
+    mechMask.mask._.g._.push if edgeBbox? then { use: { 'xlink:href': "##{edge}" } } else bboxRect null, '#fff'
```
Comment (mcous) [inline src/build-board.coffee]: Lines should not be more than 80 characters

File: src/build-board.coffee (hunk 1 above)
Comment (kasbah) [inline src/build-board.coffee]: Sorry, I am not quite sure what to do with those. They are all opened three lines above but they should not be closed on one line? Is it just that they should have spaces between them?

**Comment by kasbah** [comment]
I have made the style changes suggested.

You are correct about the clip-path but unfortunately, while it fixes rendering in Inkscape it does break it for my copy of Firefox (Chrome is fine). Reformatted copy of output here: http://kaspar.h1x.com/top.svg

I will try and report this to Inkscape too but it does happen in both 0.48 and the nightly so it's likely best to workaround as well.

**Comment by kasbah** [comment]
Nevermind, I figured out the problem (`svg-path` vs `svgPath`). This now looks perfect across all my browsers and in Inkscape too!

**Comment by mcous** [comment]
Good stuff! Want to fix up the rest of the issues and throw that clipPath stuff in?

**Comment by kasbah** [comment]
Done, it's one commit. Can split it up again if you like.

**Comment by mcous** [comment]
The clip isn't working. It should only make the copper that is not covered by mask gold, but it's making all copper gold. Watch how the whole board color changes when you change the finish color and compare that with the live site.

I did a little digging, and it turns out a `<use>` in a `<clipPath>` must reference a basic shape; referencing a `<g>` like the whole soldermask won't work. See http://codepen.io/mcous/pen/vOgKXM and [this section of the SVG spec](http://www.w3.org/TR/SVG/masking.html#EstablishingANewClippingPath)

**Comment by mcous** [comment]
Just saw the PS / PPS on your last comment, glad you noticed (and sorry for the duplication). I appreciate your work, and I'd love to get better Inkscape compatibility merged in whenever it's ready!

**Comment by kasbah** [comment]
So, the [Inkscape bug](https://bugs.launchpad.net/inkscape/+bug/794472) has been around since 2011 and the grouping workaround is mentioned there somewhere. 

Regarding the finish, I am a bit lost, it seems like the masking way should work but does not and we can't work around it using a clipPath because we break spec. The clipPath workaround does work in Inkscape actually (see updated commits), there was more confusion between `clip-path` (the property) and `clipPath` the tag.

**Comment by kasbah** [comment]
I am closing this for now, opening another PR with only the first improvement (so I can refer to this later).

**Pull request closed without merging**